File: tools/apigen/use_fidget.c
#include "stdio.h"
#include "stdbool.h"
#include "fidget.h"

// Test Fidget
int count;
void click_cb(){
    count ++;
    printf("count: %i\n", count);
}

void display_cb(){
    char countStr[20];
    snprintf(countStr, 20, "%i", count);
    Node n = fidget_find_node("text");
    fidget_node_set_characters(n, countStr);
    fidget_node_set_dirty(n, true);
}

void main(){

    count = 0;

    fidget_add_cb(E_ON_CLICK, 100, "/CounterFrame/Count1Up", click_cb);
    fidget_add_cb(E_ON_DISPLAY, 100, "/CounterFrame/CounterDisplay", display_cb);

    fidget_start_fidget(
        "https://www.figma.com/file/Km8Hvdw4wZwEk6L1bN4RLa",
        "C Counter",
        "CounterFrame",
        false
    );
}

File: tools/apigen/use_fidget.py
from fidget import *

# Test fidget
count = 0
@c_proc_cb
def click_cb():
    global count
    count += 1
    print("count:", count)
add_cb(E_ON_CLICK, 100, "/CounterFrame/Count1Up", click_cb)

@c_proc_cb
def display_cb():
    n = find_node("text")
    if n:
        n.characters = str(count)
        n.dirty = True
add_cb(E_ON_DISPLAY, 100, "/CounterFrame/CounterDisplay", display_cb)

start_fidget(
  "https://www.figma.com/file/Km8Hvdw4wZwEk6L1bN4RLa",
  "Python Counter",
  "CounterFrame",
  False
)
